File: DTLA_RT/src/filtSignal.cpp
#include <math.h>
#include <stddef.h>
#include <iostream>
#include "filtSignal.hpp"

using namespace std;

void filtSignal::filter (float *outSig,float *filtCoeff,int filtLen,float *inSig,int numSamples)
{

  float tmp =0;
  int j;

  for(int k=0;k<numSamples;k++)
  {
      for(int i=0;i<filtLen;i++)
      {
          j=k-i;
          if(j>0)
            tmp += filtCoeff[i]*inSig[j];
      }
      outSig[k] = tmp;
  }

}

File: MATLAB/DTLA_CHANNELS_NEW/filtSignal.m
function [ outSig ] = filtSignal( filtCoeff,filtLen,inSig,numSamples)


for k=1:numSamples
    tmp = 0;
    for i=1:filtLen
        j=k-i;
        if (j>0)
            tmp = tmp + filtCoeff(i)*inSig(j);
        end
    end
    outSig(k) = tmp;
end

end

